rmw_cyclonedds_cpp/src/serdata.cpp

```diff
@@ -685,11 +749,351 @@ static std::string get_type_name(const char * type_support_identifier, void * ty
+  dds_dynamic_type_kind_t kind, const char *name, uint32_t num_bounds, const uint32_t *bounds, dds_dynamic_type_t type)
+{
+  return{kind,name,{},{},num_bounds,bounds,DDS_DYNAMIC_TYPE_SPEC(type),{},};
+}
+
+extern "C" dds_dynamic_member_descriptor_t get_dynamic_member_descriptor_prim(
+  dds_dynamic_type_kind_t type, const char *name)
+{
+  return DDS_DYNAMIC_MEMBER_PRIM(type, name);
+}
+
+extern "C" dds_dynamic_member_descriptor_t get_dynamic_member_descriptor(
+  dds_dynamic_type_t ddt, const char *name)
+{
+  return DDS_DYNAMIC_MEMBER(ddt, name);
+}
+
+
+template<typename MemberType>
+static void dynamic_type_add_array(
+  dds_dynamic_type_t * dstruct, dds_entity_t dds_ppant, const MemberType * member, const dds_dynamic_type_kind_t type)
+{
+  dds_dynamic_type_t ddt;
+
+  if (member->array_size_) 
+  { 
+    if (!member->is_upper_bound_)
+    {
+      ddt = dds_dynamic_type_create(dds_ppant, 
+        get_dynamic_type_descriptor_prim(DDS_DYNAMIC_ARRAY, member->name_, 1, (uint32_t*)(&member->array_size_), type));
+    } else {
+      ddt = dds_dynamic_type_create(dds_ppant, 
+        get_dynamic_type_descriptor_prim(DDS_DYNAMIC_SEQUENCE, member->name_, 1, (uint32_t*)(&member->array_size_), type)); 
+    }
+  } else {
+    ddt = dds_dynamic_type_create(dds_ppant,
+       get_dynamic_type_descriptor_prim(DDS_DYNAMIC_SEQUENCE, member->name_, 0, nullptr, type)); 
+  }
+
+  dds_return_t ret = dds_dynamic_type_add_member(dstruct, get_dynamic_member_descriptor(ddt, member->name_));
+  if (ret != DDS_RETCODE_OK)
+      RCUTILS_LOG_ERROR("failed to add type member sequence/array with prim type");
+}
+
+template<typename MemberType>
+static void dynamic_type_add_array(
+  dds_dynamic_type * dstruct, dds_entity_t dds_ppant, const MemberType * member, const dds_dynamic_type_t ddt)
+{
+  dds_dynamic_type_t dseq;
+
+  if (member->array_size_) 
+  { 
+    if (!member->is_upper_bound_)
+    {
+      dseq = dds_dynamic_type_create(dds_ppant, 
+       get_dynamic_type_descriptor(DDS_DYNAMIC_ARRAY, member->name_, 1, (uint32_t*)(&member->array_size_), ddt));
+    } else {
+      dseq = dds_dynamic_type_create(dds_ppant, 
+        get_dynamic_type_descriptor(DDS_DYNAMIC_SEQUENCE, member->name_, 1, (uint32_t*)(&member->array_size_), ddt));
+    }
+  } else {
+    dseq = dds_dynamic_type_create(dds_ppant,
+     get_dynamic_type_descriptor(DDS_DYNAMIC_SEQUENCE, member->name_, 0, nullptr, ddt)); 
+  }
+
+  dds_return_t ret = dds_dynamic_type_add_member(dstruct, get_dynamic_member_descriptor(dseq, member->name_));
+  if (ret != DDS_RETCODE_OK)
+    RCUTILS_LOG_ERROR("failed to add type member sequence/array with dynamic type");
+}
+
+template<typename MemberType>
+static void dynamic_type_add_member(
+    dds_dynamic_type_t * dstruct, dds_entity_t dds_ppant, const MemberType * member, const dds_dynamic_type_kind_t type)
+{
+  dds_return_t ret;
+  if (type == rosidl_typesupport_introspection_cpp::ROS_TYPE_STRING ||
```

Comment:
If I am not mistaken, these two cases are handled by `construct_dds_dynamic_type`, but then I would prefer an `assert` or `abort` instead of returning without doing anything.

Reply:
I agree to continue working after errors in these functions it is not correct way.